scipy/optimize/tests/test_minpack.py

```diff
@@ -517,6 +517,26 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_covariance(self):
```

Comment:
In other words `cholesky` always computes `A = X X^T`, where `X` is lower triangular. It just returns `X^T` by default (don't why it is done like that).